### PR TITLE
fix(es/codegen): Fix codegen of some non-ascii characters

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue (if exists):**

 - Closes https://github.com/swc-project/swc/issues/5072

### CODE REVIEW

File: crates/swc_ecma_transforms_react/tests/jsx/fixture/issue-5072/output.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { Button, Input } from "antd";
+import Child from "./component/Child";
+class Page extends React.Component {
+    render() {
+        return /*#__PURE__*/ React.createElement("div", {
+            className: "test"
+        }, /*#__PURE__*/ React.createElement("div", null, "Page"), /*#__PURE__*/ React.createElement(Child, null), /*#__PURE__*/ React.createElement("input", {
+            placeholder: "我是谁?"
```
[the remark this answers is not in the document]
Comment (hardfist) [reply crates/swc_ecma_transforms_react/tests/jsx/fixture/issue-5072/output.mjs:10]: https://facebook.github.io/jsx/#sec-jsx-string-characters according to jsx spec, jsx shouldn't contains escape sequence in jsx attributes

[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_transforms_react/tests/jsx/fixture/issue-5072/output.mjs:10]: It does not transform.

[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_transforms_react/tests/jsx/fixture/issue-5072/output.mjs:10]: Problem is `codegen`, in case it's configured for es5 target

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_codegen

**Pull request merged**